TACC/Core-Portal
# FP-1650: Test Cases for _common/Button

## Overview: ##

Add test cases for button since [refining it during FP-1491 styling](https://github.com/TACC/Core-Portal/pull/598).

## Related Jira tickets: ##

* [FP-1650](https://jira.tacc.utexas.edu/browse/FP-1650)

## Summary of Changes: ##

- export and use type map and size map
- test cases for Button complexities

## Testing Steps: ##
1. Verify UI is unchanged — see https://github.com/TACC/Core-Portal/pull/598
2. Verify test cases pass.

## UI Photos:

| Top | Bottom | Misc. States |
| - | - | - |
| ![Top](https://user-images.githubusercontent.com/62723358/168701121-7effe3cb-756e-4587-b527-7b8f7d1eb0b3.png) | ![Bottom](https://user-images.githubusercontent.com/62723358/168701119-7a9022ad-57cf-4cf7-b611-0e2df63fec78.png) | ![States](https://user-images.githubusercontent.com/62723358/168701118-c84707e0-330e-4877-ad28-6ac898bc3902.png) |
